BrowserKit/Sources/SummarizeKit/SummarizeController.swift

```diff
@@ -0,0 +1,314 @@
+            summaryView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -UX.summaryViewEdgePadding),
+            summaryView.topAnchor.constraint(equalTo: closeButton.bottomAnchor, constant: UX.summaryViewEdgePadding),
+            summaryView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+
+            tabSnapshot.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            tabSnapshot.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            tabSnapshot.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+    }
+
+    private func embedSnapshot() {
+        tabSnapshot.image = viewModel.tabSnapshot
+        tabSnapshotTopConstraint?.constant = viewModel.tabSnapshotTopOffset
+
+        let frameHeight = view.frame.height
+        loadingLabel.transform = CGAffineTransform(translationX: 0.0, y: frameHeight * 0.25)
+        loadingLabel.startShimmering(light: .white, dark: .white.withAlphaComponent(0.1))
+
+        let transformAnimation = CABasicAnimation(keyPath: "transform.translation.y")
+        transformAnimation.fromValue = 0
+        transformAnimation.toValue = frameHeight / 2
+        transformAnimation.duration = UX.initialTransformAnimationDuration
+        transformAnimation.timingFunction = UX.initialTransformTimingCurve
+        transformAnimation.fillMode = .forwards
+        tabSnapshot.layer.add(transformAnimation, forKey: "translation")
+        tabSnapshot.transform = CGAffineTransform(translationX: 0.0,
+                                                  y: view.frame.height * UX.tabSnapshotInitialTransformPercentage)
+
+        gradient.animatePositionChange(animationCurve: UX.initialTransformTimingCurve)
+
+        UIView.animate(withDuration: UX.initialTransformAnimationDuration, delay: 0.0, options: [], animations: {
+            self.tabSnapshot.layer.cornerRadius = self.screenCornerRadius
+            self.loadingLabel.alpha = 1.0
+        }) { _ in
+            // This is here just to show case the full animation
+            DispatchQueue.main.asyncAfter(deadline: .now() + 4.0) {
+                self.showSummary()
+            }
+        }
+    }
+
+    private func showSummary() {
+        let theme = themeManager.getCurrentTheme(for: currentWindowUUID)
+        tabSnapshot.isUserInteractionEnabled = true
+        tabSnapshot.addGestureRecognizer(UIPanGestureRecognizer(target: self, action: #selector(onTabSnapshotPan)))
+        let tabSnapshotOffset = tabSnapshotTopConstraint?.constant ?? 0.0
+
+        UIView.animate(withDuration: UX.showSummaryAnimationDuration) { [self] in
+           gradient.alpha = 0.0
+           tabSnapshot.transform = CGAffineTransform(translationX: 0.0,
+                                                     y: view.frame.height -
+                                                     UX.tabSnapshotFinalPositionBottomPadding -
+                                                     tabSnapshotOffset)
```

Comment:
Would read better IMO.
```suggestion
           let exampleName = view.frame.height - UX.tabSnapshotFinalPositionBottomPadding - tabSnapshotOffset
           tabSnapshot.transform = CGAffineTransform(translationX: 0.0,
                                                     y: exampleName)
```